src/test_help/canned_histories.rs

```diff
@@ -253,6 +254,126 @@ pub fn single_failed_activity(activity_id: &str) -> TestHistoryBuilder {
     t
 }
 
+/// 1: EVENT_TYPE_WORKFLOW_EXECUTION_STARTED
+/// 2: EVENT_TYPE_WORKFLOW_TASK_SCHEDULED
+/// 3: EVENT_TYPE_WORKFLOW_TASK_STARTED
+/// 4: EVENT_TYPE_WORKFLOW_TASK_COMPLETED
+/// 5: EVENT_TYPE_ACTIVITY_TASK_SCHEDULED
+/// 6: EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED
+/// 7: EVENT_TYPE_WORKFLOW_TASK_SCHEDULED
+/// 8: EVENT_TYPE_WORKFLOW_TASK_STARTED
+/// 9: EVENT_TYPE_WORKFLOW_TASK_COMPLETED
+/// 10: EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED
+/// 12: EVENT_TYPE_WORKFLOW_TASK_SCHEDULED
+/// 13: EVENT_TYPE_WORKFLOW_TASK_STARTED
```

Comment:
Like we talked about both these histories probably need to be like:

```
/// 10: EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED
/// 11: EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED
```

Worth asking Max about the probable "cheat" in the Java tests

Reply:
It actually doesn't work if I replace last workflow task started/scheduled with workflow completion. Let's follow up on this.

Reply:
This will require some thought to test, as the server wouldn't bother responding with this history anyway. We can do in a different PR.